library/src/main/java/com/heinrichreimersoftware/materialintro/app/IntroActivity.java

```diff
@@ -356,36 +356,36 @@ else if (position < currentPosition && Math.ceil(position) != currentPosition &&
         animator.start();
     }
 
-    private long calculateScrollDuration(int distance) {
+   private long calculateScrollDuration(int distance) {
         return Math.round(pageScrollDuration * (distance + Math.sqrt(distance)) / 2);
     }
     
    public void goToFirstSlide() {
-        int currentItem = pager.getCurrentItem();
+        int nearestPositionToFirstSlide = pager.getCurrentItem();
         boolean canGo = false;
```

Comment:
OK, to be honest, `canGo` isn't even needed. You can just check if `pager.getCurrentItem() == nearestPositionToFirstSlide`. If they are equal, you don't have to move.
However you have to keep track of whether you were able to go the whole way to the start meaning you have to check for `nearestPositionToFirstSlide == 0`.